goalc/make/Tools.cpp

```diff
@@ -15,7 +15,7 @@
 CompilerTool::CompilerTool(Compiler* compiler) : Tool("goalc"), m_compiler(compiler) {}
 
 bool CompilerTool::needs_run(const ToolInput& task) {
-  if (!m_compiler->knows_object_file(std::filesystem::path(task.input).stem().u8string())) {
+  if (!m_compiler->knows_object_file(std::filesystem::path(task.input.at(0)).stem().u8string())) {
```

Comment:
it might be a good idea to have an error if you provide multiple inputs to a tool that just looks at the first one.